pkg/proxy/endpoints.go

```diff
@@ -396,6 +399,9 @@ func (em EndpointsMap) apply(ect *EndpointChangeTracker, staleEndpoints *[]Servi
 
 	changes := ect.checkoutChanges()
 	for _, change := range changes {
+		if ect.onEndpointsMapChange != nil {
+			ect.onEndpointsMapChange(change.previous, change.current)
```

Comment:
probably just pass change as parameter?

Reply:
probably worth beef up the comment for the apply function with something like: apply triggers processEndpointMapChange function on every change.

Reply:
The endpointChanges struct is not a public type. I will keep it that way. Any way it has just the two maps. So  no information is missed in the change handler. I will add a comment and change the function  name.

Reply:
SGTM